src/main/java/frc/robot/Robot.java

```diff
@@ -337,47 +259,136 @@ public void testPeriodic() {
     teleopPeriodic();
   }
 
+  // retrieve heading from encoder; convert it from radians to degrees
+  public double getHeadingDeg() {
+    return (m_yawEncoder.getDistance()* 180.0 / Math.PI) / 10.0;
```

Comment:
I'm wondering if we want to convert to using radians as it seems we have to convert all over the place.

Reply:
The divide by 10 is a constant I don't understand....

Reply:
Does the yaw encoder give you deci-radians?

Reply:
In any case, although we should look into it, it probably doesn't affect this PR which should contain no functionality changes.

Reply:
I suspect the problem lies here:
```
    m_yawEncoder = new Encoder(dioEncoderChanA, dioEncoderChanB, true, EncodingType.k4X);
    final double kRad2RPM = 2 * Math.PI;
    final double kEncoderTicksPerRev = 128.0 * kRad2RPM; //grayhill encoder
    m_yawEncoder.setDistancePerPulse(360.0 / kEncoderTicksPerRev);
```

Reply:
It's setting a distance per pulse in degrees, so `getDistance` should return a value in degrees. What makes you think that radians are involved?

Reply:
Oh, you're telling it to use radians right above that. But the grayhill encoder doesn't know anything about radians or degrees; only number of edges (k4x) that have passed. I guess I'm confused at why you switched from degrees to radians? (That said, swerve seems to standardize on radians, so maybe we should retain that unit.)

Reply:
I'm not convinced `kEncoderTickersPerRev` is correct, given `EncodingType.k4X`. The latter means that all four edge transitions are counted, but according to the Grayhill datasheet, one cycle is a single channel's full wave distance, which includes two of its own transitions and two transitions of the other channel; thus we should be mutliplying by 512 rather than 128, I think.

Reply:
This is assuming that we in fact have a Grayhill 63R128-080 (where that 128 in the model name means 128 ticks per cycle)